### DISCUSSION TITLE
PklError: Unable to find `pkl` command on PATH in Swift project using `apple/pkl-swift` package

### DISCUSSION THREAD

**Discussion by willswire** [apple/pkl#654] [Ideas]
Hello everyone,

I’ve encountered an issue with the `apple/pkl-swift` package when compiling a Swift project. Specifically, I’m seeing the following error:

```
PklError(message: "Unable to find `pkl` command on PATH.")
```

My expectation is that Swift projects utilizing the `apple/pkl-swift` package should not need the system to have `pkl` installed separately. Since this Swift package should handle the toolchain internally, the compiled tool should not rely on the system's PATH to locate the `pkl` binary.

Has anyone else run into this issue? Could this be related to the package configuration or how it integrates with system dependencies? I’d appreciate any insights or possible workarounds for this problem.

Thank you in advance!

**Comment by bioball** [comment]
At some point, we plan on providing a C library for Pkl so you can use Pkl without needing to spawn an external process, or need it to exist on `PATH`. For now, this is a necessary requirement for pkl-swift to work.
